{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "c1a4d9d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Test case variables \n",
    "\n",
    "col_j = \"cAa/cGa:203|Caa/Aaa:140|cAa/cTa:46|caA/caT:14|caA/caC:13|ggACaa/ggCAaa:2|cAa/cCa:2|Caa/Taa:1|CAa/AGa:1\"\n",
    "col_k = \"1:115256529_252|1:115256530_143|1:115256528_27\"\n",
    "col_a = \"NRAS\" \n",
    "col_b = \"61\"\n",
    "col_e = \"Q:422\"\n",
    "col_i = \"R:204\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "225869a9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('G', 1)]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def colj(col_j):\n",
    "    # Splits the input string by '|' and ':' characters, and extracts relevant data.\n",
    "    split_j = col_j.split(\"|\")\n",
    "    items = []\n",
    "    for i in range(0, len(split_j)):\n",
    "        split_ji = split_j[i].split(\":\")\n",
    "        a = split_ji[0].split(\"/\")[0]  \n",
    "        b = split_ji[0].split(\"/\")[1]  \n",
    "        c = split_ji[1]\n",
    "        items.append((a, b, c))\n",
    "    return items\n",
    "\n",
    "\n",
    "def loc_cap_string(s):\n",
    "    # Finds uppercase characters in the input string and their positions.\n",
    "    items = []\n",
    "    if len(s) > 3:\n",
    "        return [(\"N/A\", 0)]\n",
    "    for i in range(0, len(s)):\n",
    "        if s[i].isupper():\n",
    "            items.append((s[i], i))\n",
    "    return items\n",
    "\n",
    "\n",
    "x = colj(col_j)[0][1]  # Extract the second element from the first tuple in the result of colj()\n",
    "loc_cap_string(x)  # Call loc_cap_string() with x as input\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "4b18737e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['1', [('115256528', 0), ('115256529', 1), ('115256530', 2)]]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def kcol(col_k):\n",
    "    # Split the input string by '|' to get individual elements\n",
    "    split_k = col_k.split(\"|\")\n",
    "    # Split the first element of split_k by ':' to get the chromosome and positions\n",
    "    split_ki = split_k[0].split(\":\")\n",
    "\n",
    "    # Extract the chromosome from the first element of split_ki\n",
    "    ch_r = split_ki[0]\n",
    "\n",
    "    items = []\n",
    "    pos = 0\n",
    "    small = []\n",
    "    for i in split_k:\n",
    "        # Extract the position before the '_' character\n",
    "        small.append(i.split(\":\")[1].split(\"_\")[0])\n",
    "\n",
    "    for i in range(0, len(small)):\n",
    "        # Append each position along with its corresponding index to the items list\n",
    "        items.append((sorted(small)[i], pos))\n",
    "        pos += 1\n",
    "    \n",
    "    return [ch_r, items]\n",
    "\n",
    "kcol(col_k)  # Call kcol() with col_k as input"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "c968eb68",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'NRAS'"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def acol(col_a):\n",
    "    return \n",
    "\n",
    "col_a #Returns gene name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "c140e415",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['NRAS,1,115256529,115256529,61,Q:422,R:204,A,G,203',\n",
       " 'NRAS,1,115256528,115256528,61,Q:422,R:204,C,A,140',\n",
       " 'NRAS,1,115256529,115256529,61,Q:422,R:204,A,T,46',\n",
       " 'NRAS,1,115256530,115256530,61,Q:422,R:204,A,T,14',\n",
       " 'NRAS,1,115256530,115256530,61,Q:422,R:204,A,C,13',\n",
       " 'NRAS,1,115256528,115256528,61,Q:422,R:204,N/A,N/A,2',\n",
       " 'NRAS,1,115256529,115256529,61,Q:422,R:204,A,C,2',\n",
       " 'NRAS,1,115256528,115256528,61,Q:422,R:204,C,T,1',\n",
       " 'NRAS,1,115256528,115256529,61,Q:422,R:204,CA,AG,1']"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def finalfunc(col_a, col_b, col_e, col_i, col_j, col_k):\n",
    "    list_of_items = []\n",
    "    c_itemlist = []\n",
    "    \n",
    "    # Iterate over colj(col_j)\n",
    "    for xi in range(0, len(colj(col_j))):\n",
    "        r_itemlist = []\n",
    "        \n",
    "        # Iterate over loc_cap_string(colj(col_j)[xi][0])\n",
    "        for i in loc_cap_string(colj(col_j)[xi][0]):\n",
    "            # Iterate over kcol(col_k)[1]\n",
    "            for j in kcol(col_k)[1]:\n",
    "                if i[1] == j[1]:\n",
    "                    r_itemlist.append((i, j[0]))\n",
    "\n",
    "        a_itemlist = []\n",
    "        \n",
    "        # Iterate over loc_cap_string(colj(col_j)[xi][1])\n",
    "        for i in loc_cap_string(colj(col_j)[xi][1]):\n",
    "            # Iterate over kcol(col_k)[1]\n",
    "            for j in kcol(col_k)[1]:\n",
    "                if i[1] == j[1]:\n",
    "                    a_itemlist.append((i, j[0]))\n",
    "                    \n",
    "        c_itemlist.append(colj(col_j)[xi][2])\n",
    "            \n",
    "        ref = []\n",
    "        pos = 0\n",
    "        start = \"\"\n",
    "        \n",
    "        # Extract the uppercase characters from r_itemlist\n",
    "        for i in r_itemlist:    \n",
    "            ref.append(r_itemlist[pos][0][0])\n",
    "            pos += 1\n",
    "            start = r_itemlist[0][1]\n",
    "\n",
    "        alt = []\n",
    "        pos = 0\n",
    "        end = \"\"\n",
    "        \n",
    "        # Extract the uppercase characters from a_itemlist\n",
    "        for i in a_itemlist:    \n",
    "            alt.append(a_itemlist[pos][0][0])\n",
    "            pos += 1\n",
    "            if len(a_itemlist) > 1:\n",
    "                end = a_itemlist[len(a_itemlist)-1][1]\n",
    "            else:\n",
    "                end = a_itemlist[0][1]\n",
    "                \n",
    "        ch_r = kcol(col_k)[0]\n",
    "        gene = col_a\n",
    "        amino_acid_position = str(col_b)\n",
    "        ref_amino_acid = str(col_e)\n",
    "        alt_amino_acid = str(col_i)\n",
    "        count = c_itemlist[xi]\n",
    "        \n",
    "        # Construct the item string and append to the list_of_items\n",
    "        item = gene + \",\" + ch_r + \",\" + start + \",\" + end + \",\" + amino_acid_position + \",\" + ref_amino_acid + \",\" + alt_amino_acid + \",\" + \"\".join(ref) + \",\" + \"\".join(alt) + \",\" + count\n",
    "        list_of_items.append(item)\n",
    "        \n",
    "    return list_of_items\n",
    "\n",
    "finalfunc(col_a, col_b, col_e, col_i, col_j, col_k)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "d3de8989",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>amino_acid_position</th>\n",
       "      <th>ref_amino_acid</th>\n",
       "      <th>alt_amino_acid</th>\n",
       "      <th>ref_nt</th>\n",
       "      <th>alt_nt</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NRAS</td>\n",
       "      <td>1</td>\n",
       "      <td>115256529</td>\n",
       "      <td>115256529</td>\n",
       "      <td>61</td>\n",
       "      <td>Q:422</td>\n",
       "      <td>R:204</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NRAS</td>\n",
       "      <td>1</td>\n",
       "      <td>115256528</td>\n",
       "      <td>115256528</td>\n",
       "      <td>61</td>\n",
       "      <td>Q:422</td>\n",
       "      <td>R:204</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NRAS</td>\n",
       "      <td>1</td>\n",
       "      <td>115256529</td>\n",
       "      <td>115256529</td>\n",
       "      <td>61</td>\n",
       "      <td>Q:422</td>\n",
       "      <td>R:204</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NRAS</td>\n",
       "      <td>1</td>\n",
       "      <td>115256530</td>\n",
       "      <td>115256530</td>\n",
       "      <td>61</td>\n",
       "      <td>Q:422</td>\n",
       "      <td>R:204</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NRAS</td>\n",
       "      <td>1</td>\n",
       "      <td>115256530</td>\n",
       "      <td>115256530</td>\n",
       "      <td>61</td>\n",
       "      <td>Q:422</td>\n",
       "      <td>R:204</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NRAS</td>\n",
       "      <td>1</td>\n",
       "      <td>115256528</td>\n",
       "      <td>115256528</td>\n",
       "      <td>61</td>\n",
       "      <td>Q:422</td>\n",
       "      <td>R:204</td>\n",
       "      <td>N/A</td>\n",
       "      <td>N/A</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NRAS</td>\n",
       "      <td>1</td>\n",
       "      <td>115256529</td>\n",
       "      <td>115256529</td>\n",
       "      <td>61</td>\n",
       "      <td>Q:422</td>\n",
       "      <td>R:204</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NRAS</td>\n",
       "      <td>1</td>\n",
       "      <td>115256528</td>\n",
       "      <td>115256528</td>\n",
       "      <td>61</td>\n",
       "      <td>Q:422</td>\n",
       "      <td>R:204</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NRAS</td>\n",
       "      <td>1</td>\n",
       "      <td>115256528</td>\n",
       "      <td>115256529</td>\n",
       "      <td>61</td>\n",
       "      <td>Q:422</td>\n",
       "      <td>R:204</td>\n",
       "      <td>CA</td>\n",
       "      <td>AG</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   gene chr      start        end amino_acid_position ref_amino_acid  \\\n",
       "0  NRAS   1  115256529  115256529                  61          Q:422   \n",
       "1  NRAS   1  115256528  115256528                  61          Q:422   \n",
       "2  NRAS   1  115256529  115256529                  61          Q:422   \n",
       "3  NRAS   1  115256530  115256530                  61          Q:422   \n",
       "4  NRAS   1  115256530  115256530                  61          Q:422   \n",
       "5  NRAS   1  115256528  115256528                  61          Q:422   \n",
       "6  NRAS   1  115256529  115256529                  61          Q:422   \n",
       "7  NRAS   1  115256528  115256528                  61          Q:422   \n",
       "8  NRAS   1  115256528  115256529                  61          Q:422   \n",
       "\n",
       "  alt_amino_acid ref_nt alt_nt count  \n",
       "0          R:204      A      G   203  \n",
       "1          R:204      C      A   140  \n",
       "2          R:204      A      T    46  \n",
       "3          R:204      A      T    14  \n",
       "4          R:204      A      C    13  \n",
       "5          R:204    N/A    N/A     2  \n",
       "6          R:204      A      C     2  \n",
       "7          R:204      C      T     1  \n",
       "8          R:204     CA     AG     1  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Export to CSV\n",
    "\n",
    "formatted_list = []\n",
    "\n",
    "# Iterate over the items returned by finalfunc() and split them by comma\n",
    "for i in range(0, len(finalfunc(col_a, col_b, col_e, col_i, col_j, col_k))):\n",
    "    formatted_list.append(finalfunc(col_a, col_b, col_e, col_i, col_j, col_k)[i].split(\",\"))\n",
    "\n",
    "# Create a DataFrame with column names and data from formatted_list\n",
    "df = pd.DataFrame(columns=['gene', 'chr', 'start', 'end', 'amino_acid_position', 'ref_amino_acid', 'alt_amino_acid', 'ref_nt', 'alt_nt', 'count'], data=formatted_list)\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "466a52ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>amino_acid_position</th>\n",
       "      <th>ref_amino_acid</th>\n",
       "      <th>alt_amino_acid</th>\n",
       "      <th>ref_nt</th>\n",
       "      <th>alt_nt</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936091</td>\n",
       "      <td>545</td>\n",
       "      <td>E:633</td>\n",
       "      <td>K:575</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>3</td>\n",
       "      <td>178936092</td>\n",
       "      <td>178936092</td>\n",
       "      <td>545</td>\n",
       "      <td>E:633</td>\n",
       "      <td>K:575</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936091</td>\n",
       "      <td>545</td>\n",
       "      <td>E:633</td>\n",
       "      <td>K:575</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>3</td>\n",
       "      <td>178936092</td>\n",
       "      <td>178936092</td>\n",
       "      <td>545</td>\n",
       "      <td>E:633</td>\n",
       "      <td>K:575</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>3</td>\n",
       "      <td>178936093</td>\n",
       "      <td>178936093</td>\n",
       "      <td>545</td>\n",
       "      <td>E:633</td>\n",
       "      <td>K:575</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1484</th>\n",
       "      <td>CTCF</td>\n",
       "      <td>16</td>\n",
       "      <td>67644738</td>\n",
       "      <td>67644738</td>\n",
       "      <td>1</td>\n",
       "      <td>M:4</td>\n",
       "      <td>V:2</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1485</th>\n",
       "      <td>CTCF</td>\n",
       "      <td>16</td>\n",
       "      <td>67644737</td>\n",
       "      <td>67644737</td>\n",
       "      <td>1</td>\n",
       "      <td>M:4</td>\n",
       "      <td>V:2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1486</th>\n",
       "      <td>STK11</td>\n",
       "      <td>19</td>\n",
       "      <td>1220494</td>\n",
       "      <td>1220494</td>\n",
       "      <td>196</td>\n",
       "      <td>G:6</td>\n",
       "      <td>V:4</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1487</th>\n",
       "      <td>STK11</td>\n",
       "      <td>19</td>\n",
       "      <td>1220495</td>\n",
       "      <td>1220495</td>\n",
       "      <td>196</td>\n",
       "      <td>G:6</td>\n",
       "      <td>V:4</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1488</th>\n",
       "      <td>STK11</td>\n",
       "      <td>19</td>\n",
       "      <td>1220493</td>\n",
       "      <td>1220493</td>\n",
       "      <td>196</td>\n",
       "      <td>G:6</td>\n",
       "      <td>V:4</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1489 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene chr      start        end amino_acid_position ref_amino_acid  \\\n",
       "0     PIK3CA   3  178936091  178936091                 545          E:633   \n",
       "1     PIK3CA   3  178936092  178936092                 545          E:633   \n",
       "2     PIK3CA   3  178936091  178936091                 545          E:633   \n",
       "3     PIK3CA   3  178936092  178936092                 545          E:633   \n",
       "4     PIK3CA   3  178936093  178936093                 545          E:633   \n",
       "...      ...  ..        ...        ...                 ...            ...   \n",
       "1484    CTCF  16   67644738   67644738                   1            M:4   \n",
       "1485    CTCF  16   67644737   67644737                   1            M:4   \n",
       "1486   STK11  19    1220494    1220494                 196            G:6   \n",
       "1487   STK11  19    1220495    1220495                 196            G:6   \n",
       "1488   STK11  19    1220493    1220493                 196            G:6   \n",
       "\n",
       "     alt_amino_acid ref_nt alt_nt count  \n",
       "0             K:575      G      A   575  \n",
       "1             K:575      A      G    21  \n",
       "2             K:575      G      C    20  \n",
       "3             K:575      A      C    12  \n",
       "4             K:575      G      T     3  \n",
       "...             ...    ...    ...   ...  \n",
       "1484            V:2      G      T     1  \n",
       "1485            V:2      T      C     1  \n",
       "1486            V:4      G      T     4  \n",
       "1487            V:4      C      T     1  \n",
       "1488            V:4      G      C     1  \n",
       "\n",
       "[1489 rows x 10 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read the CSV file into a DataFrame\n",
    "df_cancer = pd.read_csv(\"data.csv\")\n",
    "\n",
    "# Display the first few rows of the DataFrame\n",
    "df_cancer.head()\n",
    "\n",
    "# Access specific columns of the DataFrame\n",
    "hugo_symbol = df_cancer['Hugo_Symbol']  # Column 'Hugo_Symbol' - variable a\n",
    "amino_acid_position = df_cancer['Amino_Acid_Position']  # Column 'Amino_Acid_Position' - variable b\n",
    "reference_amino_acid = df_cancer['Reference_Amino_Acid']  # Column 'Reference_Amino_Acid' - variable e\n",
    "variant_amino_acid = df_cancer['Variant_Amino_Acid']  # Column 'Variant_Amino_Acid' - variable i\n",
    "codon_change = df_cancer['Codon_Change']  # Column 'Codon_Change' - variable j\n",
    "genomic_position = df_cancer['Genomic_Position']  # Column 'Genomic_Position' - variable k\n",
    "\n",
    "# Create an empty list to store formatted data\n",
    "formatted_list = []\n",
    "\n",
    "# Iterate over rows in the DataFrame\n",
    "for i in range(0, len(codon_change)):\n",
    "    # Call finalfunc() with values from the corresponding row\n",
    "    result = finalfunc(hugo_symbol[i], amino_acid_position[i], reference_amino_acid[i], variant_amino_acid[i], codon_change[i], genomic_position[i])\n",
    "    # Split the result by comma and append to the formatted list\n",
    "    for j in range(0, len(result)):\n",
    "        formatted_list.append(result[j].split(\",\"))\n",
    "\n",
    "# Create a new DataFrame with formatted data\n",
    "df = pd.DataFrame(columns=['gene', 'chr', 'start', 'end', 'amino_acid_position', 'ref_amino_acid', 'alt_amino_acid', 'ref_nt', 'alt_nt', 'count'], data=formatted_list)\n",
    "\n",
    "# Display the DataFrame\n",
    "df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "722f1137",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Export the DataFrame to an Excel file\n",
    "df.to_excel(\"forward_results.xlsx\", index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
